File: Javascript/344 - Reverse String.js
/**
Write a function that reverses a string. The input string is given as an array of characters char[].
Do not allocate extra space for another array, you must do this
by modifying the input array in-place with O(1) extra memory.
You may assume all the characters consist of printable ascii characters.
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    let c = "";
    var midPoint = Math.ceil((s.length /2));
    for(var index = 0; index < midPoint; index++){
        c = s[s.length - (1 + index)];
        s[s.length - (1 + index)] = s[index];
        s[index] = c;
    }
};

File: Python/344 - Reverse String.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        listSize = ceil(len(s) / 2)
        for i in range(listSize):
            tempVal = s[i]
            s[i] = s[(len(s) - 1) - i]
            s[(len(s) - 1) - i] = tempVal
        

        """
        Write a function that reverses a string.
        The input string is given as an array of characters char[].

        Do not allocate extra space for another array,
        you must do this by modifying the input array in-place with O(1) extra memory.

        Do not return anything, modify s in-place instead.
        """
